apps/src/applab/Exporter.js

```diff
@@ -216,68 +225,94 @@ function extractCSSFromHTML(el) {
-        zipPath: appName + '/applab/common.css'
       },
     ];
+    if (expoMode) {
+      staticAssets.push({
+        url: 'https://code.jquery.com/' + jQueryBaseName + '.js',
+      });
+    }
+
+    const rootRelativeAssetPrefix = expoMode ? '' : 'assets/';
+    const zipAssetPrefix = appName + '/assets/';
 
     const appAssets = dashboard.assets.listStore.list().map(asset => ({
       url: assetPrefix.fixPath(asset.filename),
-      rootRelativePath: 'assets/' + asset.filename,
-      zipPath: appName + '/assets/' + asset.filename,
+      rootRelativePath: rootRelativeAssetPrefix + asset.filename,
+      zipPath: zipAssetPrefix + asset.filename,
       dataType: 'binary',
       filename: asset.filename,
     }));
 
-    function rewriteAssetUrls(data) {
-      return appAssets.reduce(function (data, assetToDownload) {
-        if (data.indexOf(assetToDownload.url) >= 0) {
-          return data.split(assetToDownload.url).join(assetToDownload.rootRelativePath);
-        }
-        return data.split(assetToDownload.filename).join(assetToDownload.rootRelativePath);
-      }, data);
-    }
+    const mainProjectFilesPrefix = appName + (expoMode ? '/assets/' : '/');
 
     var zip = new JSZip();
-    zip.file(appName + "/README.txt", readme);
-    zip.file(appName + "/index.html", rewriteAssetUrls(html));
-    zip.file(appName + "/style.css", rewriteAssetUrls(css));
-    zip.file(appName + "/code.js", rewriteAssetUrls(code));
+    if (expoMode) {
+      const appJson = exportExpoAppJsonEjs({
+        appName: appName,
+        projectId: project.getCurrentId()
+      });
+
+      zip.file(appName + "/package.json", exportExpoPackageJson);
+      zip.file(appName + "/app.json", appJson);
+      zip.file(appName + "/App.js", exportExpoAppJs);
+      zip.file(appName + "/CustomAsset.js", exportExpoCustomAssetJs);
+    }
+    // NOTE: for expoMode, it is important that index.html comes first...
+    zip.file(mainProjectFilesPrefix + "index.html", rewriteAssetUrls(appAssets, html));
+    zip.file(mainProjectFilesPrefix + "README.txt", readme);
+    zip.file(mainProjectFilesPrefix + "style.css", rewriteAssetUrls(appAssets, css));
+    zip.file(mainProjectFilesPrefix + (expoMode ? "code.j" : "code.js"), rewriteAssetUrls(appAssets, code));
+
+    const rootApplabPrefix = expoMode ? 'assets/applab/' : 'applab/';
+    const rootRelativeApplabAssetPrefix = rootApplabPrefix + 'assets';
+    const zipApplabAssetPrefix = appName + '/' + rootRelativeApplabAssetPrefix;
 
     return new Promise((resolve, reject) => {
       $.when(...[...staticAssets, ...appAssets].map(
         (assetToDownload) => download(assetToDownload.url, assetToDownload.dataType || 'text')
       )).then(
         ([commonLocale], [applabLocale], [applabApi], [applabCSS], [commonCSS], ...rest) => {
-          zip.file(appName + "/applab/applab-api.js",
+          zip.file(appName + "/" + (expoMode ? "assets/applab-api.j" : rootApplabPrefix + "applab-api.js"),
```

Comment:
Should this be `.js`, not just `.j`?

Reply:
This is specifically because if we include the "website assets javascript files" as `.js`, the react native bundler (which is sort of like webpack) doesn't know to ignore them and treat them as "generic" assets. So, I have to do this convoluted thing of using a different extension.